packages/shared_preferences/lib/shared_preferences.dart

```diff
@@ -45,6 +45,9 @@ class SharedPreferences {
   /// in sync since the setter method might fail for any reason.
   final Map<String, Object> _preferenceCache;
 
+  /// returns the entire set for debugging purposes.
+  Map<String, Object> getAll() => _preferenceCache;
```

Comment:
This breaks encapsulation of the class as we expose a modifiable sub-object. Client code updating this map may not be aware of `_prefix` or the fact that a platform channel has to be invoked to make the change persistent.

Debugging should be possible based on getting just a snapshot the set of keys, or alternatively a snapshot of the map (though then you'd probably want it typed `Map<String, dynamic>` for convenience). I suggest adding instead
```dart
Set<String> getKeys() => new Set<String>.from(_preferenceCache.keys);
```
Then you can still make use of the type-safety offered by the remaining methods.

Reply:
Sounds great! I'll make the change to this pull request in a few hours.